### PR TITLE
Add use_glob to GCSObjectExistenceSensor

### PR DESCRIPTION
**Pull request by A-Costa**
Initial draft implementing #32650

Add the boolean parameter `use_glob` to `GCSObjectExistenceSensor`. 

When set to `True` the sensor will use the `hook.list` method leveraging the newly introduced `match_glob` parameter to check if files are present, making it possible to define a`GCSObjectExistenceSensor` that checks if files that satisfy the given glob are present in a certain bucket.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Can you please add unit tests for it?

**Comment by A-Costa** [comment]
Sure, that was just an initial draft. I will try to push a complete version of this PR by the end of the day

File: airflow/providers/google/cloud/triggers/gcs.py [hunk 1]
```diff
@@ -98,9 +102,14 @@ async def _object_exists(self, hook: GCSAsyncHook, bucket_name: str, object_name
         async with ClientSession() as s:
             client = await hook.get_storage_client(s)
             bucket = client.get_bucket(bucket_name)
-            object_response = await bucket.blob_exists(blob_name=object_name)
-            if object_response:
-                return "success"
+            if self.use_glob:
+                object_response = await bucket.list_blobs(match_glob=object_name)
```
Comment (A-Costa) [inline airflow/providers/google/cloud/triggers/gcs.py]: this depends on https://github.com/talkiq/gcloud-aio/pull/634

**Review by shahar1 — COMMENTED** [review]
Overall LGTM :)
I hope that `gcloud-aio` will merge your PR there soon, so you can also merge it here. 
Don't forget to add a test for `GCSBlobTrigger`

**Comment by potiuk** [comment]
Needs fixes though.

**Comment by A-Costa** [comment]
https://github.com/talkiq/gcloud-aio/pull/634 was merged today so this can be fixed and merged.

Apart from the missing tests, what else needs to be fixed? @potiuk

**Comment by eladkal** [comment]
> Apart from the missing tests, what else needs to be fixed?

I think that is all but we will take closer look in 2nd review once everything is ready

**Comment by potiuk** [comment]
> any update on this?

Well. The PR is  waiting for your actions:

>  Needs fixes though

> I think that is all but we will take closer look in 2nd review once everything is ready
> You'll need also to bump gcloud-aio version in provider.yaml so it will include the fix in the upstream library


Tests are failed and @eladkal asked you to update libraries.

Why do you think you need any update before following these comments ?

**Comment by potiuk** [comment]
Generallly - making PR green is a prerequisite for anyone to take further look - if tests are failing and you were told to fix them and nothing happened, well, then the ball is on your side.

**Comment by potiuk** [comment]
(plus you have still WIP in name - usualy that is an indication that this is Work In Progress - I usually just skip over such PRs  even if they are rebased by author, because they are well, Work in Progress

**Comment by eladkal** [comment]
Happy to review when you mark the PR as ready :)
Sometimes we do multiple runs of reviews that is very OK

**Comment by potiuk** [comment]
> Here i asked if something else needed fix

Well. The CI tells you what needs to be fixed. I think you should take a look at the contributor's guide and understand that fixing and rebasing CI is what is the "first pass". You have one PR to submit. We regularly review 100s of PR a day. And CI and tests are the way to automate thing that must be fixed before we do next pass of reviews. So my comment - while it had been expressed with brevity - reflect the state -of matter. you need to fix tests before we move any furtther.

In case it is not clear - it's all described in our contributor's guide:

* How to communicate (where it is written that it's author's responsiblity to mae sure their PR gets reviewed - including periodoc rebasing and followin up on it) https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#how-to-communicate

* How to prepare PR: https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#step-4-prepare-pr
* How to iterate on PR: https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#step-5-pass-pr-review

I understand it's a lot of information to digest, so I perfectly understand your lack of understanding how our PR process looks like and what is expected from author, but now that you got it explained in a little more than "tests need fixing" I hope you will continue contributing and follow those guidelines.

I think also what helps is understand that it's you who want to contribute something to the project, so it's a good idea to pay attention to your own PR and if in doubt consult the documentation.

Looking forward to your future contributions. No hard feelings from my side, I hope with those explanations you will understand why the ball was on your side all the time, you just did not follow up.

**Comment by potiuk** [comment]
Apologies if MY comments were also seen as condescent (which I realise they might) - but this is simply coming from the sheer volume of things we have to deal with - it's very difficult to balance brevity (not helpful) with long messages (TLDR) and the time it needs to put, so we rely a lot on the docs we put together on contributing, but - yeah - it's difficult to follow, I understand that.

**Comment by A-Costa** [comment]
I've added the missing test for GCSBlobTrigger and removed WIP from the title, as this is ready to be reviewed.


> You'll need also to bump `gcloud-aio` version in `provider.yaml` so it will include the fix in the upstream library

@eladkal regarding this, it seems to me that the version is not pinned in the yaml, so it should be fine already: https://github.com/apache/airflow/blob/f7f71836175b81484fe6afb147a58e1ca6d00f4d/airflow/providers/google/provider.yaml#L90

**Comment by potiuk** [comment]
Is there a minimum version of gcloud-aio needed for that ? if so we should set >= MIN

**Comment by A-Costa** [comment]
Yes, the `match_glob` parameter is first exposed in the `9.0.0` release:
https://github.com/talkiq/gcloud-aio/releases/tag/storage-9.0.0

I'll add the minimum version

**Review by potiuk — APPROVED** [review]

**Pull request merged**